client/packages/coldchain/src/Monitoring/ListView/TemperatureBreach/TemperatureBreachList.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { FC } from 'react';
+import { useUrlQueryParams } from '@common/hooks';
+import { useTranslation } from '@common/intl';
+import {
+  CircleAlertIcon,
+  DataTable,
+  Formatter,
+  NothingHere,
+  TableProvider,
+  createTableStore,
+  useColumns,
+  useTheme,
+} from '@openmsupply-client/common';
+import {
+  TemperatureBreachFragment,
+  useTemperatureBreach,
+} from '../../api/TemperatureBreach';
+import { BreachTypeCell } from '../../../common';
+
+const ListView: FC = () => {
+  const { data, isLoading, isError } = useTemperatureBreach.document.list();
+  const {
+    updateSortQuery,
+    updatePaginationQuery,
+    // filter,
+    queryParams: { sortBy, page, first, offset },
+  } = useUrlQueryParams({ filterKey: 'startDatetime' });
+
+  const pagination = { page, first, offset };
+  const t = useTranslation('coldchain');
+  const theme = useTheme();
+
+  const columns = useColumns<TemperatureBreachFragment>(
+    [
+      {
+        key: 'acknowledgedIcon',
+        Cell: ({ rowData }) => {
+          return !rowData?.acknowledged ? (
+            <CircleAlertIcon
+              fill={theme.palette.error.main}
+              sx={{ color: '#fff' }}
+            />
+          ) : null;
+        },
+      },
+      {
+        key: 'acknowledged',
+        label: 'label.status',
+        accessor: ({ rowData }) => {
+          return !!rowData?.acknowledged
+            ? t('label.acknowledged')
+            : t('label.unacknowledged');
+        },
+        sortable: false,
+      },
+      {
+        key: 'sensor',
+        label: 'label.sensor-name',
+        accessor: ({ rowData }) => rowData.sensor?.name,
+        sortable: false,
+      },
+      {
+        key: 'locationName',
+        label: 'label.location',
+        accessor: ({ rowData }) => rowData.location?.name,
+        sortable: false,
+      },
+      {
+        key: 'cce',
+        label: 'label.cce',
+        sortable: false,
+      },
+      {
+        key: 'startDatetime',
+        label: 'label.type-start',
+        accessor: ({ rowData }) => {
+          return Formatter.csvDateTimeString(rowData.startDatetime);
+        },
+      },
+      {
+        key: 'endDatetime',
+        label: 'label.type-end',
+        accessor: ({ rowData }) => {
+          return Formatter.csvDateTimeString(rowData.endDatetime);
+        },
+      },
+      {
+        key: 'duration',
+        label: 'label.duration',
+        accessor: ({ rowData }) => {
+          return Formatter.milliseconds(rowData.duration);
+        },
+      },
+      {
+        key: 'breach',
+        label: 'label.type',
+        accessor: ({ rowData }) => rowData?.type,
+        Cell: BreachTypeCell,
+        sortable: false,
+      },
+      {
+        key: 'temperature',
+        label: 'label.temperature',
+        accessor: ({ rowData }) => {
+          return `${rowData.temperatureLog?.temperature}${t(
+            'label.temperature-unit'
+          )}`;
+        },
```

Comment:
I had to do this:

```suggestion
        accessor: ({ rowData }) => {
          return `${rowData.sensor?.latestTemperatureLog?.nodes[0]
            ?.temperature}${t('label.temperature-unit')}`;
        },
```

maybe I have bad data?

Reply:
There were some foreign key constraints during sync.. have fixed this hopefully!